python/cuspatial/cuspatial/tests/binpreds/test_equals_only_binops.py

```diff
@@ -0,0 +1,570 @@
+    gpdline1 = gpd.GeoSeries(lhs)
+    gpdline2 = gpd.GeoSeries(
+        [
+            LineString([(0, 0), (1, 1)]),
+            LineString([(1, 1), (2, 2)]),
+            LineString([(2, 2), (3, 3)]),
+        ]
+    )
+    line1 = cuspatial.from_geopandas(gpdline1)
+    line2 = cuspatial.from_geopandas(gpdline2)
+    got = line1.geom_equals(line2)
+    expected = gpdline1.geom_equals(gpdline2)
+    pd.testing.assert_series_equal(expected, got.to_pandas())
+
+
+def test_10000_linestrings_geom_equals_10000_linestrings(linestring_generator):
+    gpdlines1 = gpd.GeoSeries([*linestring_generator(10000, 5)])
+    gpdlines2 = gpd.GeoSeries([*linestring_generator(10000, 5)])
+    lines1 = cuspatial.from_geopandas(gpdlines1)
+    lines2 = cuspatial.from_geopandas(gpdlines2)
+    got = lines1.geom_equals(lines2)
+    expected = gpdlines1.geom_equals(gpdlines2)
+    pd.testing.assert_series_equal(expected, got.to_pandas())
+
+
+def test_linestring_geom_equals_polygon():
+    gpdline = gpd.GeoSeries([LineString([(0, 0), (1, 1)])])
+    gpdpolygon = gpd.GeoSeries(Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]))
+    line = cuspatial.from_geopandas(gpdline)
+    polygon = cuspatial.from_geopandas(gpdpolygon)
+    got = line.geom_equals(polygon)
+    expected = gpdline.geom_equals(gpdpolygon)
+    pd.testing.assert_series_equal(expected, got.to_pandas())
+
+
+def test_10000_linestrings_geom_equals_10000_polygons(
+    polygon_generator, linestring_generator
+):
+    gpdlines = gpd.GeoSeries([*linestring_generator(10000, 5)])
+    gpdpolygons = gpd.GeoSeries([*polygon_generator(10000, 0)])
+    lines = cuspatial.from_geopandas(gpdlines)
+    polygons = cuspatial.from_geopandas(gpdpolygons)
+    got = lines.geom_equals(polygons)
+    expected = gpdlines.geom_equals(gpdpolygons)
+    pd.testing.assert_series_equal(expected, got.to_pandas())
+
+
+def test_polygon_geom_equals_linestring():
+    gpdline = gpd.GeoSeries([LineString([(0, 0), (1, 1)])])
+    gpdpolygon = gpd.GeoSeries(Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]))
+    line = cuspatial.from_geopandas(gpdline)
+    polygon = cuspatial.from_geopandas(gpdpolygon)
+    got = polygon.geom_equals(line)
+    expected = gpdpolygon.geom_equals(gpdline)
+    pd.testing.assert_series_equal(expected, got.to_pandas())
+
+
+def test_10000_polygons_geom_equals_10000_linestrings(
+    polygon_generator, linestring_generator
+):
+    gpdpolygons = gpd.GeoSeries([*polygon_generator(10000, 0)])
+    gpdlines = gpd.GeoSeries([*linestring_generator(10000, 5)])
+    polygons = cuspatial.from_geopandas(gpdpolygons)
+    lines = cuspatial.from_geopandas(gpdlines)
+    got = polygons.geom_equals(lines)
+    expected = gpdpolygons.geom_equals(gpdlines)
+    pd.testing.assert_series_equal(expected, got.to_pandas())
+
+
+def test_polygon_geom_equals_polygon():
+    gpdpolygon1 = gpd.GeoSeries(Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]))
+    gpdpolygon2 = gpd.GeoSeries(Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]))
```

Comment:
Similarly,
```suggestion
    gpdpolygon1 = gpd.GeoSeries(Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]))
    gpdpolygon2 = gpd.GeoSeries(Polygon([[0, 0], [1, 1], [1, 0],  [0, 0]]))
```

Reply:
This example fails! I'm working on a fix.

Reply:
I've discovered that it has to do with the closed format - polygons that don't share end points don't have the same points, because the end point is duplicated. I'm working on stripping off the extra endpoint now.

Reply:
This is fixed.